docs/neat_getlpaddrs.md

```diff
@@ -0,0 +1,32 @@
+# neat_getlpaddrs
+
+Obtains the local or peer addresses of a flow.
+
+### Syntax
+
+```c
+int neat_getlpaddrs(struct neat_ctx*  ctx,
+                    struct neat_flow* flow,
+                    struct sockaddr** addrs,
+                    const int         local)
+```
+
+### Parameters
+
+- **ctx**: Pointer to a NEAT context.
+- **flow**: Pointer to a NEAT flow.
+- **addrs**: Pointer to variable for storing pointer to addresses to.
```

Comment:
Single-homing and multi-homing are already handled the same way. The differentiation is between SCTP and non-SCTP. For SCTP, sctp_getladdrs()/sctp_getpaddrs() are used to get possibly multiple addresses. The return value is the number of addresses, or -1 in case of error. sctp_getladdrs()/sctp_getpaddrs() will (most likely) not work with TCP and UDP. So, getsockname()/getpeername() are used for non-SCTP stacks. In this case, the new function just returns one address (i.e. 1 as result, or -1 in case of error). For the user of neat_getlpaddrs(), it does not matter whether the flow is single or multi-homed. The result will always be a new pointer to 1 or more addresses (the number is returned, or -1 in case of error).